src/script/common/c_content.cpp

```diff
@@ -1516,6 +1516,17 @@ void push_flags_string(lua_State *L, FlagDesc *flagdesc, u32 flags, u32 flagmask
 	lua_pushlstring(L, flagstring.c_str(), flagstring.size());
 }
 
+void push_flags_table(lua_State *L, FlagDesc *flagdesc, u32 flags, u32 flagmask)
+{
+	lua_newtable(L);
+
+	for (int i = 0; flagdesc[i].name; i++) {
+		if (flagmask & flagdesc[i].flag) {
+			setboolfield(L, -1, flagdesc[i].name, !!(flags & flagdesc[i].flag));
```

Comment:
You already know that it's true.
`setboolfield(L, -1, flagdesc[i].name, true);`

Reply:
the if clause checks `flagmask`, not `flags`